lib/cloud_controller/uaa/uaa_token_decoder.rb

```diff
@@ -7,12 +7,13 @@ class BadToken < StandardError
 
     attr_reader :config
 
-    def initialize(uaa_config, grace_period_in_seconds=0)
+    def initialize(uaa_config, grace_period_in_seconds=0, alternate_reference_time=nil)
```

Comment:
What about adding a constraint that only one of the properties `grace_period_in_seconds` and `alternate_reference_time` can be set? I don't think that it makes much sense to have an alternate reference time and a grace period. So if both are provided (i.e. != 0 and != nil), an exception could be raised.